docs/reference/query-dsl/random-sample-query.asciidoc

```diff
@@ -0,0 +1,77 @@
+[role="xpack"]
+[[query-dsl-random-sample-query]]
+=== Random sample query
+++++
+<titleabbrev>Random sample</titleabbrev>
+++++
+
+experimental::[]
+
+Wraps a <<query-dsl-bool-query, filter query>> and returns a random subset of all the
+matching documents. This query is best used expensive <<search-aggregations, aggregations>> over large datasets.
+It can provide a significant speed improvement at the cost of loss of accuracy.
+
+This query does not score documents and shouldn't be used outside of pairing it with aggs. Additionally, sampling may
+provide poor results when used with a very restrictive `filter`.
+
+[source,console]
+----
+GET kibana_sample_data_ecommerce/_search?size=0&track_total_hits=false
+{
+  "query": {
+    "random_sample": {
+      "probability": 0.1
+    }
+  },
+  "aggregations": {
+    "price_percentiles": {
+      "percentiles": {
+        "field": "taxful_total_price"
+      }
+    }
+  }
+}
+----
+// TEST[setup:kibana_sample_data_ecommerce]
+
+[[random-sample-top-level-params]]
+==== Top-level parameters for `random_sample`
+`filter`::
++
+--
+(Optional, query object) <<query-dsl-bool-query, Filter query>> you wish to use.
+Any returned documents must match this query. If not provided, all documents are matched.
+
+Filter queries do not calculate <<relevance-scores,relevance scores>>. To
+speed up performance, {es} automatically caches frequently used filter queries.
+--
+
+`probability`::
+(Required, float) Floating point number indicating the sampling rate for the documents.
+Must be between (0, 1) exclusive. The `probability` isn't exactly the percentage of documents sampled,
+but for the majority of cases they can be considered equivalent concepts. Example, a `"probability": 0.01` over
+`1000000` documents will generally sample `100000` (or `1%`) documents. But it may not be exactly `100000` and is
+usually within the window of `[9900, 101000]`.
```

Comment:
```suggestion
but for the majority of cases, they can be considered equivalent concepts. For example, a `"probability": 0.01` over
`1000000` documents generally samples ` 100000` (or `1%`) documents. But it may not be exactly `100000` and is
usually within the window of `[9900, 101000]`.
```